mastodon.js

```diff
@@ -21,55 +21,57 @@ function get (path,params) {
 		log.info("get",path,params)
 }
 
+function genShortURL(){
+	// Were assuming single thread sequential URL generation here...
+	let num = Math.floor(Date.now()/1000)-1669485542;
+	let digits = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+	let len = Math.min(digits.length, 62);
+	let result = '';
+	while (num > 0) {
+		result = digits[num % len] + result;
+		num = parseInt(num / len, 10);
+	}
+	return result;
+}
+
 async function videoReply(filename,mediaType,replyTo,text,tweet,checksum,hasAudio,program,mode){
 
+		let short_url=genShortURL();
 
 	if (tweet.spoiler_text == ""){
 		console.log("No CW on bot source post")
 	}
 
 	try {
-		 let progData = encodeURIComponent(JSON.stringify({
-                        "v":1,
-                        "program":program,
-                        "author":text,
-                        "date": Date.now(),
-												"id": replyTo
-                }));
-
-		progData = progData.replace(/\(/g, '%28').replace(/\)/g, '%29');
 
 		let resp = await toot.post('media', { file: fs.createReadStream(filename), description:"BBC Micro Bot graphics output - "+tweet.spoiler_text });
 		log.info(resp)
 		let id = resp.data.id; // Source: https://bbcmic.ro/#"+progData
-		let params = { status:"I ran "+text+"'s program and got this.", media_ids: [id],in_reply_to_id:replyTo};
+		let params = { status:"I ran "+text+"'s program and got this.\nSource: http://link.bbcmic.ro/"+short_url+" #bbcbasic", media_ids: [id],in_reply_to_id:replyTo};
 		params.visibility = "public";
 
 		let response = await toot.post('statuses', params);
-
-		//log.info(response)
+		log.info("Media post DONE ",JSON.stringify(response));
 
 		await toot.post('statuses/:id/favourite', { id: [tweet.id]});
 		log.info("Favourited "+tweet.id);
 
-		log.info("Media post DONE ");
 
-		// let record = {
-		// 		"v":2,
-		// 		"author":tweet.user.screen_name,
-		// 		"program":program,
-		// 		"mode":mode,
-		// 		"date":Math.floor(new Date(tweet.created_at))/1000,
-		// 		"in_reply_to_id":tweet.id
-		// 		}
-		//
-		// await fs.writeFileSync('./output/'+response.id, JSON.stringify(record,null,4));
+		let progData = encodeURIComponent(JSON.stringify({
+											 "v":3,  // Mastodon era
+											 "program":program,
+											 "src": tweet.url
+							 }));
+
+	 progData = progData.replace(/\(/g, '%28').replace(/\)/g, '%29');
```

Comment:
I realise this was already there, but I don't think it's needed now.  Looking at the history this seems to have been added so the link posted to discord didn't result in broken markdown, but the discord integration has been removed since.